eppo_client/client.py

```diff
@@ -442,19 +457,34 @@ def check_value_type_match(
         return isinstance(value, bool)
     return False
 
+def convert_context_attributes_to_attributes(
+    subject_context: Union[ContextAttributes, Attributes]
+) -> Attributes:
+    if isinstance(subject_context, dict):
+        return subject_context
+    
+    return subject_context.numeric_attributes | subject_context.categorical_attributes
+
 
-def convert_subject_context_to_attributes(
+def convert_subject_context_to_context_attributes(
     subject_context: Union[ContextAttributes, Attributes]
 ) -> ContextAttributes:
     if isinstance(subject_context, dict):
         return ContextAttributes.from_dict(subject_context)
-    return subject_context
+
+    stringified_categorical_attributes = {
+        key: str(value) for key, value in subject_context.categorical_attributes.items()
+    }
```

Comment:
@lisaah's original fix, which is to stringify the values of all explicitly passed in categorical attributes so they can be evaluated correctly against the model

Reply:
Should we also ensure conversion to string when subject_context is a dictionary?

Reply:
I guess in that case we don't know what values should be converted, so bugs can still arise but we have no way to fix.

Reply:
`ContextAttributes.from_dict()` handles this... bucketing all numeric values numeric attributes and then the string values of strings and bools as categorical attributes